### DISCUSSION TITLE
Sprite "Shake" in one Container.

### DISCUSSION THREAD

**Discussion by neciszhang** [pixijs/pixijs#7598] [Q&A] [answered]
### Expected Behavior
I want the sprite don't "shake" when run. The problem is here [video](https://three.tzjjtech.cn/1626326323237565.mp4);
Can you give me ideas?
### Environment

- **`pixi.js` version**: _e.g. 6.0_
- **Browser & Version**: _e.g. Chrome 91_
- **OS & Version**: _e.g. Mac OS 10.15.7
- **Running Example**: [link](https://three.tzjjtech.cn/car/#/home)

**Comment by neciszhang** [comment]
The running code is here:
play(delta) {
      //road running
      road.tilePosition.y += speed;
      //oils running
      if (oils.length >= 3) {
        if (oils[0]) {
          oils[0].y += speed;
        }
        if (oils[1]) {
          oils[1].y += speed;
        }
        if (oils[2]) {
          oils[2].y += speed;
        }
      }
      //create oils
      this.createOil();
      //remove the oil out of canvas 
      this.removeOutOil();
    },

**Comment by ivanpopelyshev** [comment]
Does shaking start only after a few minutes?

**Comment by ivanpopelyshev** [comment, marked as answer]
Oh, right, tilingSprite ... you should mod it regularly, or you get precision problem on shader

```js
road.tilePosition.y %= road.texture.height;
```

**Reply by neciszhang** [reply]
Hi, Geek
I upload a l [link](https://three.tzjjtech.cn/car/#/home).
I try to change the code.I failed.

**Reply by ivanpopelyshev** [reply]
Your javascript is obfuscated, I cannot read it.

Read this: https://github.com/pixijs/pixijs/issues?q=is%3Aissue+tilingsprite+precision

**Reply by neciszhang** [reply]
Hello, i upload the code.
[main code](https://three.tzjjtech.cn/Home.vue)
[full code](https://three.tzjjtech.cn/code1.zip)

**Reply by ivanpopelyshev** [reply]
So, whats the problem here?

```js
    // let next = defaultY+100*361;
    // if (road.tilePosition.y >= next) {
    //   let cha= road.tilePosition.y-next;
    //   console.log(cha);
    //   road.tilePosition.y=defaultY+cha;
    // }else{
    //   road.tilePosition.y += speed;
    // }
 ```

just reduce it sometimes by value that is divisible by `tilingSprite.texture.height`.

This bug stays for years, so far no one made a PR for it , even many people complained ;) I dont want to do it because its good learning experience for all coders who start with pixi.

**Reply by neciszhang** [reply]
Thank you very much! I find the problem.
